### PR TITLE
Fix broken links & sync documentation

### PR DESCRIPTION
**Pull request by sent**
- Fixed broken links (Edit this page on Github link on each page & some page links)
- Synced documentation.
- Added static redirection of old URLs.
- Added trailing slashes to all URLs (to improve correct redirection from users.html to /users/ otherwise weird problems with gh pages).
- Removed $ for easier copying.

### CODE REVIEW

**Comment by msimerson** [comment]
done? read to merge?

**Comment by sent** [comment]
Hi @msimerson,

Sorry for not replying faster. There is a missing feature that allows nuxt content having different repositories content & code spitted up.

Good news is that there is already a feature request https://github.com/nuxt/content/issues/689 which will be released in January.

Until then there are two possibilities:
- Having Github link in the header redirected to this repository? Which isn't nice.
- Having 'Edit this page on GitHub' being 404 links to Github.

Please, let me know which one would be a temporary solution for now.

Also, it's currently not navigating up to releases page via header link with trailing slashes enabled, PR with the fix will be pushed to nuxt content!

However:
http://haraka.github.io/releases -> works
http://haraka.github.io/releases/ -> works

Thank you!

**Comment by msimerson** [comment]
How would it affect things if the code for the web site were instead in the main repo as the `gh-pages` branch?

**Comment by sent** [comment]
> How would it affect things if the code for the web site were instead in the main repo as the `gh-pages` branch?

@msimerson well, the problem actually is that currently it's not possible to override those documentation edit links and it could be solved:
- only by moving this repo inside https://github.com/haraka/Haraka/tree/master/docs so there will be only one documentation *
- wait new release in January.

\* https://github.com/haraka/Haraka/edit/master/docs/content/en/community.md

I can understand this is something completely new (edge) but without giving a chance it won't progress and be better I would say 👍 

Thank for understanding and sorry for causing a little bit troubles with this!

**Comment by msimerson** [comment]
> only by moving this repo inside https://github.com/haraka/Haraka/tree/master/docs so there will be only one documentation *

Much of the documentation already exists in that location and is then periodically duplicated/synced to this repo. That's a manual step and seems like something best automated away.

If we moved the rest of the doc bits into haraka/Haraka/docs, and updated those files with the jekyll headers and metadata, besides fixing the doc edit links, would that make publishing the site much simpler in the future?

**Comment by baudehlo** [comment]
We can probably script the publishing of it from a Github Action when we
release.

**Comment by sent** [comment]
> > only by moving this repo inside https://github.com/haraka/Haraka/tree/master/docs so there will be only one documentation *
> 
> Much of the documentation already exists in that location and is then periodically duplicated/synced to this repo. That's a manual step and seems like something best automated away.
> 
> If we moved the rest of the doc bits into haraka/Haraka/docs, and updated those files with the jekyll headers and metadata, besides fixing the doc edit links, would that make publishing the site much simpler in the future?

Hi guys!

I've just tried it up, here we go:

I've copied Haraka repository to https://github.com/mailtm/Mailtm and moved this repositories files to docs folder and updated Github actions for deployment.

It's now automated deployment on each push to master it will create gh-pages branch on [**mailtm/mailtm.github.io** ](https://github.com/mailtm/mailtm.github.io).

Basically **peaceiris/actions-gh-pages** does all job as written here: https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-deploy-to-external-repository-external_repository 

Supports both personal & deployment keys.

Please let me know how we should proceed, I'm ready to work on this!

Thanks!

**Comment by msimerson** [comment]
Since you're doing the work, I'm inclined to follow your lead. My question is, which is easier to maintain?  From my perch, it seems that getting rid of the duplication and syncing of files between repos would be one less maintenance step.

It would necessitate updating `bin/haraka` so that the `-h` help feature could find the new paths to the source files (easy). Maybe we'd even add some logic to it to strip off the jekyll headers (also easy) and maybe also pass the markdown through a converter that formatted them as man pages.

**Comment by sent** [comment]
> Since you're doing the work, I'm inclined to follow your lead. My question is, which is easier to maintain? From my perch, it seems that getting rid of the duplication and syncing of files between repos would be one less maintenance step.
> 
> It would necessitate updating `bin/haraka` so that the `-h` help feature could find the new paths to the source files (easy). Maybe we'd even add some logic to it to strip off the jekyll headers (also easy) and maybe also pass the markdown through a converter that formatted them as man pages.

After much thinking I came to conclusion that maintaining this in the master branch of Haraka might be bad idea actually. It will create unneeded noise with updating package.json or yarn.lock (or anything else with it) of nuxt content.

Also all new releases will contain unneeded files of it.

I'm ready to sync documentation whenever something is changed (not much changes actually I would say).

Basically, solution would be creating a docs branch on the main repository and when changes pushed to it deploy to this repository via **peaceiris/actions-gh-pages**.

This repository is needed because Github Pages on the main repository doesn't provide root path* when deploying sadly.

\*haraka/Haraka -> deployed as haraka.github.com/Haraka/

Does it sound good now?

Thanks!
